src/tools/perf/libperf.c

```diff
@@ -1012,3 +1027,106 @@ ucs_status_t ucx_perf_run(ucx_perf_params_t *params, ucx_perf_result_t *result)
 out:
     return status;
 }
+
+
+/* multiple threads sharing the same worker/iface */
+
+typedef struct {
+    pthread_t pt;
+    int tid;
+    int ntid;
+    ucx_perf_context_t perf;
+    ucx_perf_params_t params;
+    ucx_perf_result_t result;
+} ucx_perf_thread_context_t;
+
+static ucs_status_t* ucx_perf_thread_statuses;
+static pthread_barrier_t ucx_perf_thread_barrier;
+
+static void* ucx_perf_thread_run_test(void* tctx_) {
+    ucx_perf_thread_context_t* tctx = (ucx_perf_thread_context_t*) tctx_;
+    ucx_perf_params_t* params = &tctx->params;
+    ucx_perf_result_t* result = &tctx->result;
+    ucx_perf_context_t* perf = &tctx->perf;
+    int tid = tctx->tid;
+//    int rid = rte_call(perf, group_index);
+    int i;
+
+    if (params->warmup_iter > 0) {
+        ucx_perf_set_warmup(perf, params);
```

Comment:
previously seemed like you have allocated a buffer space for each thread, do you also need to adjust the parameters here? because now seems like all threads would use the same pointers